arcgis-ios-sdk-samples/Maps/Create and save a map/CreateSaveMapViewController.swift

```diff
@@ -46,14 +63,34 @@ class CreateSaveMapViewController: UIViewController, CreateOptionsViewController
             "CreateOptionsViewController",
             "SaveAsViewController"
         ]
-        
-        // Auth Manager settings
-        let config = AGSOAuthConfiguration(portalURL: nil, clientID: "xHx4Nj7q1g19Wh6P", redirectURL: "iOSSamples://auth")
-        AGSAuthenticationManager.shared().oAuthConfigurations.add(config)
+        let portal = AGSPortal.arcGISOnline(withLoginRequired: true)
+        self.portal = portal
+        portal.load { [weak self] (error) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get the user's array of portal folders.
+                portal.user?.fetchContent { _, folders, _ in
+                    if let portalFolders = folders {
+                        self.portalFolders = portalFolders
+                    }
+                }
+                // Initially show the map creation UI.
+                self.performSegue(withIdentifier: "CreateNewSegue", sender: self)
+                self.newMapButton.isEnabled = true
+                self.saveButton.isEnabled = true
+            }
+        }
+    }
+    
+    deinit {
+        // Reset the API key after successful login.
+        AGSArcGISRuntimeEnvironment.apiKey = apiKey
+        if let config = oAuthConfiguration {
```

Comment:
I believe there would be a build [error](https://github.com/Esri/arcgis-runtime-samples-ios/pull/1086#discussion_r661878252) shown on this line.